test/integration/tracked_vehicle_system.cc

```diff
@@ -0,0 +1,367 @@
+    // at least a bit behind the cylinder (0, -1, 0). The driving is a bit wild,
+    // so we don't care much about the end Y position and yaw.
+    EXPECT_LT(poses.back().Pos().X(), -0.99);  // The driving is wild
+    EXPECT_NEAR(poses.back().Pos().Y(), 0, 0.5);
+    EXPECT_NEAR(poses.back().Pos().Z(), 0.0, 1e-1);
+    EXPECT_ANGLE_NEAR(poses.back().Rot().Roll(), 0.0, 1e-1);
+    EXPECT_ANGLE_NEAR(poses.back().Rot().Pitch(), 0.0, 1e-1);
+    // The driving is wild
+    EXPECT_ANGLE_NEAR(
+      poses.back().Rot().Yaw(), beforeCylinderPose.Rot().Yaw(), 0.5);
+
+    // Test driving over an obstacle that requires flippers. Without them, the
+    // robot would get stuck in front of the obstacle.
+
+    const auto beforeBoxPose = math::Pose3d(
+      1, 2, 0.1,
+      0, 0, -math::Angle::Pi.Radian());
+    model.SetWorldPoseCmd(*ecm, beforeBoxPose);
+
+    // Let the model settle down.
+    server.Run(true, 300, false);
+
+    // we go backwards because we have the CoG in the back
+    msgs::Set(msg.mutable_linear(), math::Vector3d(-linearSpeed, 0, 0));
+    msgs::Set(msg.mutable_angular(), math::Vector3d(0, 0, 0));
+    pub.Publish(msg);
+    server.Run(true, 4000, false);
+
+    // The box is at (2, 2, 0), we start at (1, 2, 0), and want to pass
+    // at least a bit behind the box (3.5, 2, 0). The driving is a bit wild.
+    EXPECT_GT(poses.back().Pos().X(), 3.5);
+    EXPECT_NEAR(poses.back().Pos().Y(), 2, 0.1);  // The driving is wild
+    EXPECT_NEAR(poses.back().Pos().Z(), 0.0, 1e-1);
+    EXPECT_ANGLE_NEAR(poses.back().Rot().Roll(), 0.0, 1e-1);
+    EXPECT_ANGLE_NEAR(poses.back().Rot().Pitch(), 0.0, 1e-1);
+    // The driving is wild
+    EXPECT_ANGLE_NEAR(
+      poses.back().Rot().Yaw(), beforeBoxPose.Rot().Yaw(), 0.25);
+    // And we go back, which is a somewhat easier way
+
+    msgs::Set(msg.mutable_linear(), math::Vector3d(linearSpeed, 0, 0));
+    msgs::Set(msg.mutable_angular(), math::Vector3d(0, 0, 0));
+    pub.Publish(msg);
+    server.Run(true, 4000, false);
+
+    // We start at (3.5, 2, 0), we go back, and it should be a bit faster than
+    // the previous traversal, so we should end up beyond the starting point.
+    EXPECT_LT(poses.back().Pos().X(), 1);
+    EXPECT_NEAR(poses.back().Pos().Y(), 2, 0.1);  // The driving is wild
+    EXPECT_NEAR(poses.back().Pos().Z(), 0.0, 1e-1);
+    EXPECT_ANGLE_NEAR(poses.back().Rot().Roll(), 0.0, 1e-1);
+    EXPECT_ANGLE_NEAR(poses.back().Rot().Pitch(), 0.0, 1e-1);
+    // The driving is wild
+    EXPECT_ANGLE_NEAR(
+      poses.back().Rot().Yaw(), beforeBoxPose.Rot().Yaw(), 0.25);
+  }
+};
+
+/////////////////////////////////////////////////
+TEST_P(TrackedVehicleTest, PublishCmd)
+{
+  TestPublishCmd(
+    std::string(PROJECT_SOURCE_PATH) +
+    "/test/worlds/tracked_vehicle_simple.sdf",
+    "/model/simple_tracked/cmd_vel",
+    "/model/simple_tracked/odometry");
+}
+
+// Run multiple times
+INSTANTIATE_TEST_SUITE_P(ServerRepeat, TrackedVehicleTest,
+    ::testing::Range(1, 2));
```

Comment:
Can you clarify why we're running the test multiple times?

Reply:
No particular reason. This has been only a leftover from basing the test off of `diff_drive_system.cc` test. I converted it to a normal single-shot test.

However, the pattern from the diff drive test is in a few other integration tests, but it always seems to trigger just a single run of the test (looking at Jenkins results). That's weird, and probably these other tests could also be simplified.